interview-questions.md

```diff
@@ -6,46 +6,169 @@ First, without external resources. Challenge yourself to answer from memory.
 
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
-1. What does calling super do in a Ruby class?
+---
+
+### 1. What does calling super do in a Ruby class?
 
 Your answer:
 
+In Ruby, super is like calling the same method in the parent class when you're in a subclass. It lets the subclass use the same method from the parent class and add extra behavior to it if needed. It's like borrowing something from your parent and adding your own touch to it.
+
 Researched answer:
 
-2. What is a gem?
+In Ruby, calling super in a class is used to invoke the same-named method from the superclass, allowing the subclass to inherit and extend the behavior of the parent class's method. Taking a fictional example inspired by Super Mario Bros, let's consider a Player class as the superclass and a Mario class as the subclass.
+```ruby
+class Player
+  def jump
+    puts "Player jumps"
+  end
+end
+
+class Mario < Player
+  def jump
+    puts "Mario jumps higher" 
+    super # calling the 'jump' method from the parent class
+  end
+end
+
+mario = Mario.new
+mario.jump
+```
+In this example, the Player class has a jump method that prints "Player jumps". The Mario class inherits from the Player class and overrides the jump method. The super call within the Mario class invokes the jump method from the Player class, allowing Mario to extend the behavior of the parent class's method by adding additional code after the super call.
+
+When mario.jump is called, it will output:
+```ruby
+Mario jumps higher
+Player jumps
+```
+As you can see, the jump method from the Player class is called first due to the super call in the Mario class, followed by the additional code in the Mario class's jump method. This allows Mario to inherit the behavior of the parent class's method and extend it with its own modifications.
+
```

Comment:
Nice example! Just a note: when `mario.jump` is called, the `jump` method of the `Mario` class is executed first, which is seen in the output when "Mario jumps higher" prints before "Player jumps"